marketplace/deployer_util/config_helper.py

```diff
@@ -778,6 +778,34 @@ class SchemaXServiceAccount:
 
   def __init__(self, dictionary):
     self._roles = dictionary.get('roles', [])
+    for role in self._roles:
+      if role.get('rulesType') == 'PREDEFINED':
+        if role.get('rules'):
+          raise InvalidSchema('rules can only be used with rulesType CUSTOM')
+        if not role.get('rulesFromRoleName'):
+          raise InvalidSchema('Missing rulesFromRoleName for PREDEFINED role')
+      elif role.get('rulesType') == 'CUSTOM':
+        if role.get('rulesFromRoleName'):
+          raise InvalidSchema(
+              'rulesFromRoleName can only be used with rulesType PREDEFINED')
+        if not role.get('rules'):
+          raise InvalidSchema('Missing rules for CUSTOM role')
+        for rule in role.get('rules', []):
+          if rule.get('nonResourceURLs'):
+            raise InvalidSchema(
+                'Only attributes for resourceRules are supported in rules')
+          if not rule.get('apiGroups') or not filter(lambda x: x,
+                                                     rule.get('apiGroups')):
+            raise InvalidSchema("Missing or empty apiGroups in rules. "
+                                "Did you mean [\"v1\"] or [\"*\"]?")
```

Comment:
Hi @eshiroma, this rule broke our CI command (`mpdev verify` ..) because we have empty `apiGroups`:
https://github.com/cloudbees/core-google-launcher/blob/master/schema-test.yaml#L39-L156

Per K8s documentation, it looks like `""` is an acceptable value for `apiGroups`:
https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole

So, is this validation really needed? Is there something else we might be missing?

Thanks!

cc @Vlatombe @jglick